src/App.jsx

```diff
@@ -3,46 +3,115 @@ import React, { useState, useEffect } from 'react';
 import AddTodoForm from './AddTodoForm';
 import TodoList from './TodoList';
 
-function useSemiPersistentState() {
-  const [todoList, setTodoList] = useState(
-    JSON.parse(localStorage.getItem('savedTodoList')) || []
-  );
-
-  useEffect(() => {
-    localStorage.setItem('savedTodoList', JSON.stringify(todoList));
-  }, [todoList]);
-
-  return [todoList, setTodoList];
-}
-
 function App() {
-  const [todoList, setTodoList] = useSemiPersistentState();
-
-  const addTodo = (newTodo) => {
-    setTodoList([...todoList, newTodo]);
+  
+  // Load todoList from localStorage or initialize it as an empty array
+  const storedTodoList = JSON.parse(localStorage.getItem('todoList')) || [];
+  const [todoList, setTodoList] = useState(storedTodoList);
+  const [isLoading, setIsLoading] = useState(true);
+   
+  // Function to update todoList state
+  const updateTodoList = (newTodoList) => {
+    setTodoList([...newTodoList]);
   };
 
-  const removeTodo = (id) => {
-    const updatedTodoList = todoList.filter(todo => todo.id !== id);
-    setTodoList(updatedTodoList);
-  } 
+  // Load initial data from Airtable and save to localStorage
+  useEffect(() => {
+    const fetchData = async () => {
+      try {
+        const url = `https://api.airtable.com/v0/${import.meta.env.VITE_AIRTABLE_BASE_ID}/${import.meta.env.VITE_TABLE_NAME}`;
```

Comment:
Now this same URL is being used multiple times but it doesn't need to be scoped to each block. How can you call this URL from every function that needs it.

Reply:
`const airtableURL = `https://api.airtable.com/v0/${import.meta.env.VITE_AIRTABLE_BASE_ID}/${import.meta.env.VITE_TABLE_NAME}`;`
I create a variable at the beginning and call it whenever I need it.